File: strategy_pattern/java/MiniDuckSimulator.java
import java.util.ArrayList;
import java.util.List;

public class MiniDuckSimulator {
   public static void main(String[] args) {
      List<Duck> ducks = new ArrayList<>();
      ducks.add(new MallardDuck());
      ducks.add(new RubberDuck());
      ducks.add(new RocketDuck());

      for (Duck duck : ducks) {
         duck.display();
         duck.fly();
         duck.quack();
         System.out.println("\n");
      }
   }
}

File: ruby/strategy_pattern/mini_duck_simulator.rb
# frozen_string_literal: true

require_relative 'mallard_duck'
require_relative 'rocket_duck'
require_relative 'rubber_duck'
# Provides a runner for the other classes
class MiniDuckSimulator
  @ducks = [MallardDuck.new, RubberDuck.new, RocketDuck.new]

  @ducks.each do |duck|
    duck.display
    duck.fly
    duck.quack
    puts "\n"
  end
end
